compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp

```diff
@@ -78,6 +78,33 @@ luci::CircleReshape *create_cloned_reshape(luci::CircleReshape *reshape)
   return new_reshape;
 }
 
+bool forward_reshape(luci::CircleReshape *reshape, luci::CircleMean *mean, uint32_t axis)
+{
+  assert(reshape != nullptr); // FIX_CALLER_UNLESS
+  assert(mean != nullptr);    // FIX_CALLER_UNLESS
+
+  auto new_reshape = create_cloned_reshape(reshape);
+  if (not new_reshape)
+    return false;
+
+  // reconnect network
+  loco::replace(mean).with(new_reshape);
+  mean->input(reshape->tensor());
+  new_reshape->tensor(mean);
+
+  // Change const shape axis value
+  auto *shape_reshape = loco::must_cast<luci::CircleConst *>(new_reshape->shape());
+  assert(shape_reshape->dtype() == loco::DataType::S32);     // FIX_CALLER_UNLESS
+  assert(axis < shape_reshape->size<loco::DataType::S32>()); // FIX_CALLER_UNLESS
+  shape_reshape->at<loco::DataType::S32>(axis) = 1;
```

Comment:
just one more thing for future beginners,
```suggestion
  // Mean reduction will make value to '1'
  shape_reshape->at<loco::DataType::S32>(axis) = 1;
```

Reply:
done